client/packages/common/src/hooks/useKeyboardShortcut/useKeyboardShortcut.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+
+interface props {
+  key: string;
+  handler: () => void;
+  dependencies?: any[];
+}
+
+export const useKeyboardShortcut = ({ key, handler, dependencies }: props) => {
+  useEffect(() => {
+    function handleKeyDown(this: HTMLElement, ev: KeyboardEvent) {
+      if (ev.ctrlKey && ev.key === key) {
+        handler();
+      }
+    }
+
+    document.body.addEventListener('keydown', handleKeyDown);
+    return () => document.body.removeEventListener('keydown', handleKeyDown);
+  }, dependencies);
+};
```

Comment:
We already have the "KBar" for managing keyboard shortcuts. Shouldn't we just implement all keyboard shortcuts through that rather than have two different mechanisms?

I think it would be confusing for users as to why only some of the keyboard shortcuts are listed with Cmd-K. (not to mention without being listed in there, you're unlikely to discover new shortcuts!)

Reply:
the problem is how to call an action which is on a specific page from the shared commandK instance. KBar is instantiated at a global level, whereas these keyboard actions are specific to a page / part of site.

I looked at adding an action to kbar. didn't want to pass functions around globally though.
There is `useRegisterActions` so possibly I could use that but it does mean that we can't use modifiers (the ctrl or alt) and we are already running out of keys to use!

Reply:
> There is `useRegisterActions` so possibly I could use that but it does mean that we can't use modifiers (the ctrl or alt) and we are already running out of keys to use!

Hmmm, I thought we could use `ctrl`, `alt` etc with KBar... but I can't seem to figure out how now. 😔

Reply:
Ah! I found it! It's very poorly documented, but the correct syntax in KBar is `Control+s` -- just tested it and it works with `useRegisterActions` :)